bob/src/cluster/operations.rs

```diff
@@ -26,15 +27,16 @@ fn is_result_successful<TErr: Debug>(
     0
 }
 
-pub(crate) async fn finish_at_least_handles<TErr: Debug>(
-    handles: &mut Tasks<TErr>,
+pub(crate) async fn finish_at_least_handles<TRes, TErr: Debug>(
+    handles: &mut Tasks<TRes, TErr>,
     at_least: usize,
+    get_affected_count: impl Fn(TRes) -> usize
```

Comment:
`finish_at_least` should not know anything about affected replicas count. Instead it should return 2 vectors: 
1. With successful results
2. With errors